Tasks/Common/webdeployment-common/xmlvariablesubstitutionutility.ts

```diff
@@ -1,22 +1,91 @@
 import Q = require('q');
 import tl = require('vsts-task-lib/task');
 import fs = require('fs');
+import path = require('path');
 
 var ltx = require("ltx");
 var varUtility = require ('./variableutility.js');
 var ltxdomutility = require("./ltxdomutility.js");
 var fileEncoding = require('./fileencoding.js');
 
+function getReplacableTokenFromTags(xmlNode, variableMap) {
+    var parameterSubValue = {};
+    for (var childNode of xmlNode.children) {
+        if(!varUtility.isObject(childNode)) {
+            continue;
+        }
+        for(var nodeAttributes in childNode.attrs) {
+            if (childNode.attrs[nodeAttributes].startsWith('$(ReplacableToken_') && variableMap[childNode.attrs['name']]) {
+                var lastIndexOf_ = childNode.attrs[nodeAttributes].lastIndexOf('_');
+                if(lastIndexOf_ < 19) {
```

Comment:
Don't hardcode the values (19 and 18). Calculate the length of this replaceable token and use.